src/components/remote_control/src/remote_control_plugin.cc

```diff
@@ -323,4 +347,23 @@ ResourceAllocationManager& RemoteControlPlugin::resource_allocation_manager() {
   return resource_allocation_manager_;
 }
 
+void RemoteControlPlugin::OnSDLEvent(functional_modules::SDLEvent event,
+                                     const uint32_t application_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  LOG4CXX_DEBUG(logger_, "Event " << event << " came for " << application_id);
+
+  if (functional_modules::SDLEvent::kApplicationPolicyUpdated == event) {
+    ProcessApplicationPolicyUpdate();
+    return;
+  }
+
+  Resources acquired_modules =
+      resource_allocation_manager_.GetAcquiredResources(application_id);
+
+  Resources::const_iterator module = acquired_modules.begin();
+  for (; acquired_modules.end() != module; ++module) {
+    resource_allocation_manager_.ReleaseResource(*module, application_id);
```

Comment:
@dev-gh suggest to add to `resource_allocation_manager_` function like `ReleaseAllocatedResources(app_id)` and move this loop to this new function. Then you could remove `GetAcquiredResources` + `ReleaseResource` functions.

Reply:
@AKalinich-Luxoft 
To me 'ReleaseAllocatedResources(app_id)' looks like all resources of application must be released, however idea was to release only specific resources.
Instead I came to another option: I think whole logic should be placed inside ResourceAllocationManager and OnSDLEvent interface should be added for this class so this manager will decide internally what to do on particular event coming. Also it looks like this manager class don't have to know about plugin but about service instead.

Reply:
@AKalinich-Luxoft 
Done with d05c695